src/test/java/core/basesyntax/service/operations/SupplyOperationTest.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax.service.operations;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.exceptions.InvalidFruitException;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class SupplyOperationTest {
+    private FruitTransaction transaction;
+    private final OperationHandler operation = new SupplyOperation();
+
+    @BeforeEach
+    void beforeAll() {
+        Storage.clear();
+        Storage.put(new Fruit("apple"), 10);
+    }
+
+    @Test
+    void apply_correctOperation_Ok() {
+        transaction = new FruitTransaction(
+                FruitTransaction.Operation.SUPPLY,new Fruit("apple"),1);
+        operation.apply(transaction);
+        Map<Fruit, Integer> expected = new HashMap<>();
+        expected.put(new Fruit("apple"),11);
+        Assertions.assertEquals(expected,Storage.copy());
+    }
+
+    @Test
+    void apply_invalidFruit_NotOk() {

```

Comment:
Your test case 'apply_invalidFruit_NotOk' might not be testing what you expect. Based on the setup in the @BeforeEach method, there is no 'banana' in the storage. If the 'SupplyOperation' class is intended to throw 'InvalidFruitException' when the fruit does not exist in the storage, this test case is fine. However, if the 'SupplyOperation' class is intended to add the fruit to the storage when it does not exist, this test case will fail.

Reply:
ok, done